File: client/src/app/models/data-models/questionPreferences.ts
export interface QuestionPreferences {
    questionPreferencesID: number;
    checklistTypeName: string;
    key: string;
    keyNumber?: number;
    label: string;
    questionKindID: number;
    standardQuestionID: number;
    isVisible: boolean;
    questionOrder?: number;
    minValue?: number;
    maxValue?: number;
}

export interface QuestionPrefDto {
    questionPreferencesID: number;
    isVisible: boolean;
}

File: API/Models/Tables/Entities/QuestionPreferences.cs
namespace API.Models.Tables.Entities {
    public class QuestionPreferences : BaseEntity {
        public int questionPreferencesID { get; set; }
        public int userID { get; set; }
        public int checklistTypeID { get; set; }
        public string key { get; set; }
        public int? keyNumber { get; set; }
        public string label { get; set; }
        public int questionKindID { get; set; }
        public int standardQuestionID { get; set; }
        public bool isVisible { get; set; }
        public int? questionOrder { get; set; }
        public int? minValue { get; set; }
        public int? maxValue { get; set; }
    }
}